### PR TITLE
Adds support for remote loading widget configs + hour re-renaming

### PR DESCRIPTION
**Pull request by laander**
Full support for loading widget configs from the Timekit API using the new  `/widgets` endpoints. 

Important changes that must be taken when deploying:
- Hour users embedding the widget should be notified to change the CDN address and div ID (to `#bookingjs`) (a temporary hardcoded fallback have been provided though)
- Try to redirect hour CDN calls to timekit CDN?

Dependent deploys:
- Hour app https://github.com/timekit-io/hour-app/pull/16
- Hour hosted https://github.com/timekit-io/hour-hosted/pull/1

Note: the library was renamed to `hour-widget` and renamed back again to `booking-js`

### CODE REVIEW

File: test/configLoading.spec.js [hunk 1]
```diff
@@ -0,0 +1,68 @@
+'use strict';
+
+jasmine.getFixtures().fixturesPath = 'base/test/fixtures';
+
+var baseConfig = require('./utils/defaultConfig');
```
Comment (laander) [inline test/configLoading.spec.js]: Codacy Issue found: ["baseConfig" is defined but never used](https://www.codacy.com/app/laander/booking-js/file/2533384293/issues/source?bid=2969108&fileBranchId=3284650#l5)

File: src/main.js [hunk 2]
```diff
@@ -571,17 +584,49 @@ function TimekitBooking() {
...
+  };
+
+  // Load config from remote (embed or hosted)
+  var loadRemoteConfig = function(suppliedConfig) {
+
+    config = setConfigDefaults(suppliedConfig)
+    timekitSetupConfig();
+    if (suppliedConfig.widgetId) {
+      return timekit
+      .getEmbedWidget({ id: suppliedConfig.widgetId })
```
Comment (laander) [inline src/main.js]: Codacy Issue found: ["response" is defined but never used](https://www.codacy.com/app/laander/booking-js/file/2533384286/issues/source?bid=2969108&fileBranchId=3284650#l609)

File: src/main.js [hunk 3]
```diff
@@ -449,38 +451,45 @@ function TimekitBooking() {
...
     .createBooking(args);
 
   };
 
   // Render the powered by Timekit message
   var renderPoweredByMessage = function(pageTarget) {
 
+    var campaignName = 'widget'
+    var campaignSource = window.location.hostname.replace(/\./g, '-')
+    if (config.widgetSlug) campaignName = 'hosted-widget'
```
Comment (laander) [inline src/main.js]: Codacy Issue found: [Expected '{' and instead saw 'campaignName'.](https://www.codacy.com/app/laander/booking-js/file/2533384286/issues/source?bid=2969108&fileBranchId=3284650#l473)

File: src/main.js [hunk 4]
```diff
@@ -449,38 +451,45 @@ function TimekitBooking() {
...
+    .include('attributes')
     .createBooking(args);
 
   };
 
   // Render the powered by Timekit message
   var renderPoweredByMessage = function(pageTarget) {
 
+    var campaignName = 'widget'
+    var campaignSource = window.location.hostname.replace(/\./g, '-')
```
Comment (laander) [inline src/main.js]: Codacy Issue found: [Expected '{' and instead saw 'campaignName'.](https://www.codacy.com/app/laander/booking-js/file/2533384286/issues/source?bid=2969108&fileBranchId=3284650#l472)

**Comment by vistik** [comment]
@laander maybe you should test this using the mothership :)

File: src/main.js [hunk 5]
```diff
@@ -571,17 +580,51 @@ function TimekitBooking() {
...
+  };
+
+  // Load config from remote (embed or hosted)
+  var loadRemoteConfig = function(suppliedConfig) {
+
+    config = setConfigDefaults(suppliedConfig)
+    timekitSetupConfig();
+    if (suppliedConfig.widgetId) {
+      return timekit
+      .getEmbedWidget({ id: suppliedConfig.widgetId })
```
Comment (laander) [inline src/main.js]: Codacy Issue found: ["response" is defined but never used](https://www.codacy.com/app/laander/booking-js/file/2598583893/issues/source?bid=2969108&fileBranchId=3284650#l605)

File: src/main.js [hunk 6]
```diff
@@ -449,51 +451,54 @@ function TimekitBooking() {
...
     .createBooking(args);
 
   };
 
   // Render the powered by Timekit message
   var renderPoweredByMessage = function(pageTarget) {
 
+    var campaignName = 'widget'
+    var campaignSource = window.location.hostname.replace(/\./g, '-')
+    if (config.widgetSlug) campaignName = 'hosted-widget'
```
Comment (laander) [inline src/main.js]: Codacy Issue found: [Expected '{' and instead saw 'campaignName'.](https://www.codacy.com/app/laander/booking-js/file/2598583893/issues/source?bid=2969108&fileBranchId=3284650#l473)

File: src/main.js [hunk 7]
```diff
@@ -449,51 +451,54 @@ function TimekitBooking() {
...
+    .include('attributes')
     .createBooking(args);
 
   };
 
   // Render the powered by Timekit message
   var renderPoweredByMessage = function(pageTarget) {
 
+    var campaignName = 'widget'
+    var campaignSource = window.location.hostname.replace(/\./g, '-')
```
Comment (laander) [inline src/main.js]: Codacy Issue found: [Expected '{' and instead saw 'campaignName'.](https://www.codacy.com/app/laander/booking-js/file/2598583893/issues/source?bid=2969108&fileBranchId=3284650#l472)

File: test/configLoading.spec.js (hunk 1 above)
Comment (laander) [inline test/configLoading.spec.js]: Codacy Issue found: ["baseConfig" is defined but never used](https://www.codacy.com/app/laander/booking-js/file/2598583923/issues/source?bid=2969108&fileBranchId=3284650#l5)

**Comment by vistik** [comment]
@laander There is conflics & codacy is puking  

`Hour users embedding the widget should be notified to change the CDN address and div ID (to #bookingjs) (a temporary hardcoded fallback have been provided though)` 
Does it breaking anything? if not do you need people to chance something at some stage?

You should test this using mothership

**Comment by laander** [comment]
@vistik 

Fixed merge conflicts.
Can't test using mothership as all API calls are mocked. Could be changed in the future, but I'm afraid its a time sink right now.
It won't break anything short term (I've made compatibility fallbacks), but I'll write people through Intercom letting them know that they should generate a new embed code. In Cloudfront, it's possible to get a report of "Top Referrers", so I think I could run through that list in a week and see if some sites are still hitting the old CDN url.

**Comment by laander** [comment]
Btw when i try to load the Codacy page, i just get a Chrome "Aw, Snap!" error 💩

**Comment by vistik** [comment]
🎱

**Pull request merged**